xcape/cape_fortran.py

```diff
@@ -1,14 +1,26 @@
 
-def cape(p_2d, t_2d, td_2d, p_s, t_s, td_s, source, ml_depth, adiabat, pinc, type_grid):
-    from xcape import CAPE_CODE
+def cape(p_2d, t_2d, td_2d, p_s, t_s, td_s, pres_lev_pos, source, ml_depth, adiabat, pinc, type_grid):
+    
+                
     # nlev has to be the first dimension
-
+    # nlev here is the number of levels in 3d variables (without surface level)
     nlev, ngrid = p_2d.shape
+    # type_grid  type of vertical grid: 1 for model levels, 2 for pressure levels:
+    if type_grid == 1:
+        from xcape import CAPE_CODE_model_lev
     
-    CAPE, CIN, MUlev, zMUlev = CAPE_CODE.loopcape(p_2d, t_2d, td_2d,
-                                                  p_s, t_s, td_s,
-                                                    pinc, source, ml_depth, adiabat, type_grid,
-                                                    nlev, ngrid)
-#     print(MUlev)
+        CAPE, CIN, MUlev, zMUlev = CAPE_CODE_model_lev.loopcape(p_2d, t_2d, td_2d,
+                                                      p_s, t_s, td_s,
+                                                      pinc, source, ml_depth, adiabat,
+                                                      nlev, ngrid)
```

Comment:
In CAPE_CODE_model_lev there is not pres_lev_pos because it is always 1. So it is hardcoded in the fortran code.